jenkinsbuilder/indexdindexymltransform.py

```diff
@@ -0,0 +1,196 @@
+                job_id = entry["job-id"]
+                git_url = entry["git-url"]
+                git_path = entry["git-path"]
+                git_branch = entry["git-branch"]
+                target_file = entry["target-file"]
+                desired_tag = entry["desired-tag"]
+                notify_email = entry["notify-email"]
+                depends_on = entry["depends-on"]
+
+                # Append the collected and generated information to indexyml data collector
+                index_yml["Projects"].append({
+                    "id": tid,
+                    "app-id": app_id,
+                    "job-id": job_id,
+                    "git-url": git_url,
+                    "git-path": git_path,
+                    "git-branch": git_branch,
+                    "target-file": target_file,
+                    "desired-tag": desired_tag,
+                    "notify-email": notify_email,
+                    "depends-on": depends_on
+                })
+
+                current_index += 1
+
+        # Dump the indexyml data to specfied index.yml
+        with open(self._index_yml, "w") as index_yml_file: 
+            safe_dump(index_yml, index_yml_file, default_flow_style=False, indent=2)
+
+        # Add few lines of comment indicating the the file is generated.
+        cl1 = "This file is generated by the CentOS Community Container Pipeline Service"
+        cl2 = "Please do not edit this file manually."
+        commentline = str.format("# {0}\n# {1}\n\n", cl1, cl2)
+
+        with open(self._index_yml, "r") as original:
+            data = original.read()
+
+        with open(self._index_yml, "w") as modified:
+            modified.write(commentline + data)
+
+        return
+
+    def _genindexd(self):
+        """Generate index.d files given index.yml"""
+
+        index_yml = None
+        indexd_data = {}
+
+        # Open the indexyml file and read the yaml data
+        with open(self._index_yml, "r") as index_file:
+            index_yml = load(index_file)
+
+        # For every project in the indexyml file, collect the nessasary data
+        for project in index_yml["Projects"]:
+            app_id = project["app-id"]
+            job_id = project["job-id"]
+            git_url = project["git-url"]
+            git_path = project["git-path"]
+            git_branch = project["git-branch"]
+            target_file = project["target-file"]
+            notify_email = project["notify-email"]
+            depends_on = project["depends-on"]
+
+            # If no desired-tag is specified, assume a default, otherwise take specified desired-tag
+            if "desired-tag" not in project.keys():
+                desired_tag = "latest"
+
+            else:
+                desired_tag = project["desired-tag"]
+
+            # If there isnt already indexd data entry for a specific appid, add it.
+            # The appid will be used later on to set the filenames
+            if app_id not in indexd_data.keys():
```

Comment:
The indexd_data is used to store the content to be written into the index.yml file

For example if postgres and centos are 2 appids index.yml, then
indexd_data["postgres"] = all entries in index.yml of postgres app-id
indexd_data["centos"] = all entries in index.yml of centos app-id

Then i iterate through the dict creating a ${key}.yml file and dumping ${value} which are all entries into that file so

indexd_data.key = centos hence filename = "centos.yml" and data = "value" which is yml pertaining to centos containers.